test/test_torch.py

```diff
@@ -10270,6 +10276,88 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @unittest.skipIf(not TEST_SCIPY, 'Scipy not found')
+    @dtypes(torch.float, torch.double, torch.half)
+    @dtypesIfCUDA(torch.float, torch.double, torch.half, torch.bfloat16)
```

Comment:
This looks like a repeated pattern, can it perhaps be refactored into a common wrapper?
Another random question: why `bfloat16` is CUDA-only? I thought that [bfloat16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) is native float16 CPU type, isn't it?